hazelcast/src/main/java/com/hazelcast/internal/networking/nio/SelectorOptimizer.java

```diff
@@ -42,6 +44,30 @@
     private SelectorOptimizer() {
     }
 
+    /**
+     * Creates a new Selector and will optimize it if possible.
+     *
+     * @param logger the logger used for the optimization process.
+     * @return the created Selector.
+     * @throws NullPointerException if logger is null.
+     */
+    public static Selector newSelector(ILogger logger) {
+        checkNotNull(logger, "logger");
+
+        Selector selector;
+        try {
+            selector = Selector.open();
+        } catch (IOException e) {
+            throw new HazelcastException("Failed to open a Selector", e);
+        }
+
+        boolean optimize = Boolean.parseBoolean(System.getProperty("hazelcast.io.optimizeselector", "true"));
+        if (optimize) {
+            optimize(selector, logger);
```

Comment:
Can the optimize method be `private`? Looks like an internal matter of the optimizer now.

Reply:
Good point. I'll make it package private due to tests referring to this method.